src/main/java/sim/model/Server.java

```diff
@@ -77,16 +72,22 @@ public long getSpeed() {
     // Subject methods
     // add observers to a list
 	public void registerObserver(Observer o) {
-		observers.add(o);
+		LinkedList<Observer> current = new LinkedList<>();
```

Comment:
No need to instantiate a new LinkedList here when it is immediately replaced by the cloned list after